packages/plans-grid/src/plan-item/index.tsx

```diff
@@ -91,115 +103,120 @@ function domainMessageStateMachine(
 export interface Props {
 	slug: string;
 	name: string;
+	description: string;
 	price: string;
 	features: Array< string >;
 	domain?: DomainSuggestions.DomainSuggestion;
-	isPopular?: boolean;
+	badge?: string;
 	isFree?: boolean;
+	isOpen?: boolean;
+	isPrimary?: boolean;
 	isSelected?: boolean;
 	onSelect: ( slug: string ) => void;
 	onPickDomainClick?: () => void;
-	onToggleExpandAll?: () => void;
-	allPlansExpanded: boolean;
+	onToggle?: ( slug: string, isOpen: boolean ) => void;
 }
 
 const PlanItem: React.FunctionComponent< Props > = ( {
 	slug,
 	name,
+	description,
 	price,
-	isPopular = false,
-	isFree = false,
-	domain,
 	features,
+	domain,
+	badge,
+	isFree = false,
+	isOpen = false,
+	isPrimary = false,
 	onSelect,
 	onPickDomainClick,
-	onToggleExpandAll,
-	allPlansExpanded,
+	onToggle,
 } ) => {
 	const { __ } = useI18n();
 
-	const [ isOpenInternalState, setIsOpenInternalState ] = useState( false );
-
-	const isDesktop = useViewportMatch( 'mobile', '>=' );
-
 	// show a nbps in price while loading to prevent a janky UI
 	const nbsp = '\u00A0';
 
 	const domainMessage = domainMessageStateMachine( isFree, domain, __ );
 
-	useEffect( () => {
-		setIsOpenInternalState( allPlansExpanded );
-	}, [ allPlansExpanded ] );
-
-	const isOpen = allPlansExpanded || isDesktop || isPopular || isOpenInternalState;
+	const handleToggle = () => {
+		onToggle?.( slug, ! isOpen );
+	};
 
 	return (
-		<div className={ classNames( 'plan-item', { 'is-popular': isPopular, 'is-open': isOpen } ) }>
-			{ isPopular && <span className="plan-item__badge">{ __( 'Popular' ) }</span> }
-			<div className={ classNames( 'plan-item__viewport', { 'is-popular': isPopular } ) }>
+		<div
+			className={ classNames( 'plan-item', {
+				'is-open': isOpen,
+				'is-primary': isPrimary,
+				'has-badge': !! badge,
+			} ) }
+		>
+			{ badge && (
+				<div className="plan-item__badge">
+					<span>{ badge }</span>
+				</div>
+			) }
+			<div className="plan-item__viewport">
 				<div className="plan-item__details">
 					<div
 						tabIndex={ 0 }
 						role="button"
-						onClick={ () => setIsOpenInternalState( ( open ) => ! open ) }
-						onKeyDown={ ( e ) =>
-							e.keyCode === SPACE_BAR_KEYCODE && setIsOpenInternalState( ( open ) => ! open )
-						}
-						className="plan-item__summary"
+						onClick={ handleToggle }
+						onKeyDown={ ( e ) => e.keyCode === SPACE_BAR_KEYCODE && handleToggle() }
```

Comment:
Why not use also _ENTER_ key to toggle accordion as mentioned in [WAI-ARIA spec](https://www.w3.org/TR/wai-aria-practices/)?

Reply:
And also with focus outline (not sure if there is at the moment). Another PR to fix A11y issues.